packages/tools/viewer/src/viewer.ts

```diff
@@ -981,6 +986,36 @@ export class Viewer implements IDisposable {
         await this._updateModel(undefined, undefined, abortSignal);
     }
 
+    /**
+     * Adds a 3D model from the specified URL to the scene.
+     * @remarks
+     * Will be stored in the viewer's list of loaded models.
+     * @param source A url or File or ArrayBufferView that points to the model to load.
+     * @param options The options to use when loading the model.
+     * @param abortSignal An optional signal that can be used to abort the loading process.
+     * @returns The loaded model.
+     */
+    protected async _addModel(source: string | File | ArrayBufferView, options?: LoadModelOptions, abortSignal?: AbortSignal): Promise<Model> {
+        const model = await this._loadModel(source, options, abortSignal);
+        this._loadedModelsBacking.push(model);
+        return model;
+    }
+
+    /**
+     * Removes a 3D model from the scene.
+     * @param model The model to remove.
+     */
+    protected async _removeModel(model: Model): Promise<void> {
+        const index = this._loadedModelsBacking.indexOf(model);
+        if (index !== -1) {
+            this._loadedModelsBacking.splice(index, 1);
+            if (model === this._activeModel) {
+                this._setActiveModel(null);
+            }
+            model.dispose();
+        }
+    }

```

Comment:
Instead of adding this new function, can we just do this work directly in `model.dispose` so that there is only one way to dispose and remove a model? Otherwise it seems unclear whether you are supposed to call `model.dispose()` or `viewer._removeModel`.